src/main/java/uk/gov/hmcts/reform/civil/service/PaymentRequestUpdateCallbackService.java

```diff
@@ -3,167 +3,236 @@
+    private boolean isPaid(ServiceRequestUpdateDto dto) {
+        return PAID.equalsIgnoreCase(dto.getServiceRequestStatus());
     }
 
-    private CardPaymentStatusResponse getCardPaymentStatusResponse(ServiceRequestUpdateDto serviceRequestUpdateDto) {
+    private FeeType getFeeType(String feeTypeStr) {
+        try {
+            return FeeType.valueOf(feeTypeStr);
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
 
-        return CardPaymentStatusResponse.builder().paymentReference(serviceRequestUpdateDto.getPayment().getPaymentReference())
-            .status(String.valueOf(SUCCESS)).build();
+    private boolean isHandledFeeType(FeeType feeType) {
+        return feeType == FeeType.HEARING || feeType == FeeType.CLAIMISSUED;
     }
 
-    private void createEvent(CaseData caseData, String caseId, String feeType) {
+    private boolean isPaymentUpdateValid(FeeType feeType, CaseData caseData) {
+        return (feeType == FeeType.HEARING && isHearingPaymentFailed(caseData))
+            || (feeType == FeeType.CLAIMISSUED && isClaimIssuePaymentFailed(caseData));
+    }
 
-        StartEventResponse startEventResponse = coreCaseDataService.startUpdate(
-            caseId,
-            getEventNameFromFeeType(caseData, feeType)
-        );
+    private boolean isHearingPaymentFailed(CaseData caseData) {
+        PaymentDetails hearingPayment = caseData.getHearingFeePaymentDetails();
+        return hearingPayment == null || hearingPayment.getStatus() == PaymentStatus.FAILED;
+    }
 
-        CaseDataContent caseDataContent = buildCaseDataContent(
-            startEventResponse,
-            caseData
+    private boolean isClaimIssuePaymentFailed(CaseData caseData) {
+        PaymentDetails claimIssuePayment = caseData.getClaimIssuedPaymentDetails();
+        return claimIssuePayment == null || claimIssuePayment.getStatus() == PaymentStatus.FAILED;
+    }
+
+    @Retryable(value = CaseDataUpdateException.class, backoff = @Backoff(delay = 500))
+    public void updatePaymentStatus(FeeType feeType, String caseReference, CardPaymentStatusResponse paymentStatusResponse) {
+        try {
+            Long caseId = Long.valueOf(caseReference);
+            CaseDetails caseDetails = coreCaseDataService.getCase(caseId);
+            CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
+            caseData = updateCaseDataWithPaymentDetails(paymentStatusResponse, caseData, feeType);
+
+            createAndSubmitEvent(caseData, caseId, feeType, false);
+        } catch (Exception ex) {
+            log.error("Error updating payment status for case {}: {}", caseReference, ex.getMessage());
+            throw new CaseDataUpdateException();
+        }
+    }
+
+    private void handlePaymentUpdate(ServiceRequestUpdateDto dto, CaseData caseData, FeeType feeType, Long caseId, boolean isValidPayment) {
```

Comment:
not a fan of passing booleans round like this - also don't really understand how it relates? Just do the logic in the handlePaymentUpdate method rather than passing in the boolean imo